vault/errors.go

```diff
@@ -2,10 +2,40 @@ package vault
 
 import (
 	"fmt"
+	"regexp"
 )
 
-var NotFound error
+var secretErrorRegexp, keyErrorRegexp *regexp.Regexp
+
+//IsNotFound returns true if the given error is a SecretNotFound error
+// 	or a KeyNotFound error. Returns false otherwise.
+func IsNotFound(err error) bool {
+	return isSecretNotFound(err) || isKeyNotFound(err)
+}
+
+//NewSecretNotFoundError returns an error with a message descibing the path
+// which could not be found in the secret backend.
+func NewSecretNotFoundError(path string) error {
+	return fmt.Errorf("no secret exists at path `%s`", path)
+}
+
+func isSecretNotFound(err error) bool {
+	return secretErrorRegexp.Match([]byte(err.Error()))
+}
```

Comment:
Rather than relying on regexp, can we do more OO-style Error handling, and be all idiomatically go-like?